slh-dsa/tests/pkcs8.rs

```diff
@@ -0,0 +1,39 @@
+#![cfg(feature = "alloc")]
+
+use hex_literal::hex;
+use pkcs8::{DecodePrivateKey, EncodePrivateKey, EncodePublicKey, LineEnding};
+use slh_dsa::{Sha2_128s, SigningKey, VerifyingKey};
+use std::ops::Deref;
+
+#[test]
+fn pkcs8_output() {
+    let signing = SigningKey::<Sha2_128s>::try_from(&hex!("A0FC7756572F3008F544399C25C9E087C28287AB54ADB1601FCACF85C2995A54404F690CD9A145512F61F2E4DE9292DA71371E754B3C2A79F2471E14608A2E34")[..]).unwrap();
+
+    let out = signing.to_pkcs8_pem(LineEnding::LF).unwrap();
+
+    assert_eq!(
+        out.deref(),
+        r#"-----BEGIN PRIVATE KEY-----
+MFICAQAwCwYJYIZIAWUDBAMUBECg/HdWVy8wCPVEOZwlyeCHwoKHq1StsWAfys+F
+wplaVEBPaQzZoUVRL2Hy5N6SktpxNx51SzwqefJHHhRgii40
+-----END PRIVATE KEY-----
```

Comment:
That's fine if it's been tested against multiple implementations, but perhaps add a comment about its origin?

Reply:
actually, the serialization was slightly different. The last IETF hackaton used the temporary OID assignment, this is the same ASN1 with the newer OIDs instead.

I will try to find a better one.

Reply:
FWIW I asked they include test vectors in the draft for Kyber certificates. It would be nice if there were the same for SLH-DSA: https://github.com/lamps-wg/kyber-certificates/issues/1#issuecomment-2332691055

Reply:
There is one here
https://github.com/lamps-wg/x509-slhdsa/blob/main/id-slh-dsa-sha2-128s.priv